### DISCUSSION TITLE
React Select wrapped in Controller cannot be styled

### DISCUSSION THREAD

**Discussion by king612** [react-hook-form/react-hook-form#3697] [General] [answered]
I am trying to use a React Select wrapped Component control, but Controller appears to inject a bunch of CSS on the fly that prevents normal styling (of either) using className.  This is quite annoying, so I wonder if I'm doing something wrong.

I have a drop down built as shown below.  

I'm showing the className at both levels for illustration, but at either level, it is totally ignored:

```
                    <div className="form-group col-md-2">
                        <label>Type</label>
                        <Controller className="form-control form-control-sm"
                            as={<Select className="form-control form-control-sm" />}
                            options={typeOptions}
                            name="type"
                            control={control}
                            onChange={([selected]) => {
                                return { value: selected };
                            }}
                            rules={{ required: true }}
                        />
                        { errors.type && <span className="has-error">Locus type is required</span> }
                    </div>
```

Either way, you get this hosed up rendering that looks like this:

![image](https://user-images.githubusercontent.com/5648848/102240410-d05a7280-3ec5-11eb-917a-cda8fd4e6241.png)

Presumably because of all this injected CSS:

![image](https://user-images.githubusercontent.com/5648848/102240673-1d3e4900-3ec6-11eb-8e77-9b72e99f3dd4.png)

I haven't been able to find any way around this.

Any suggestions would be greatly appreciated.

**Comment by bluebill1049** [comment, marked as answer]
you can use `render` prop, https://react-hook-form.com/api/#Controller

eg: 

```
<Controller
  control={control}
  name="test"
  render={(
    { onChange, onBlur, value, name, ref },
    { invalid, isTouched, isDirty }
  ) => (
    <Checkbox
      onBlur={onBlur}
      onChange={(e) => onChange(e.target.checked)}
      checked={value}
      inputRef={ref}
    />
  )}
/>
```

**Reply by king612** [reply]
Thank you, but this was a CSS styling issue.  Not seeing how this would solve that.

**Reply by bluebill1049** [reply]
can't you add className in the checkbox example above? this can then apply to your react-select question.

**Reply by king612** [reply]
Ah, yes.  I just wasn't sure which level it would go on (Controller or Checkbox / Select).  I'll try that - thanks!
